### DISCUSSION TITLE
Where are  the natures and Abilites being taken from?

### DISCUSSION THREAD

**Discussion by ThunderSpark91** [kwsch/PKHeX#3543] [Q&A] [answered]
I have been looking back into this program to find the issue for PKMN-NTR ability's filter. While it is true PKHex uses a older version 18.0.1.28 I would like to see in this recent one where you get the natures and abilities of Pokémon specifically then I can get further from there.

**Comment by Lusamine** [comment, marked as answer]
Information about a Pokemon species comes from personal data that comes from each of the games.

Binaries: https://github.com/kwsch/PKHeX/tree/master/PKHeX.Core/Resources/byte/personal
Structures: https://github.com/kwsch/PKHeX/tree/master/PKHeX.Core/PersonalInfo

**Reply by ThunderSpark91** [reply]
Okay that is some clarification thank you I have looked through it, but I do not see how you read the nature. Does that work differently?

I get that you search for the Ability Index, but here is some question for clarification. Even though the Pokémon might not have three abilities, I am to assume that still use all three index values? Because I noticed in PK Hex version 18.0.1.28 that a Pokémon like Entei has Pressure listed twice before you get the hidden ability. Can you explain how this actually happens? Because so far I understand that you have a bool that checks second regular ability. And just equates to it if it is false.  Can I also assume the hidden ability is index 3?
Is this assumption correct and am I reading this right? I know this is from Personal Info Black and White. But I am going to assume ORAS, XY and SM kind of work the same.

![afbeelding](https://user-images.githubusercontent.com/59621887/177985951-e759ddb6-a743-40b8-9aff-3c5129355ce1.png)




Thank you for the feedback.
Unfortunately, Pokémon NTR project is (that uses PK Hex version 18.0.1.28) sadly dead in terms of future updates. I cannot contact the owner of that project to clarify on how he reads the Nature and compares it to PK Hex. That was my idea to read it and compare. Thank you anyway.

**Comment by Lusamine** [comment]
It says Ability 1, 2, and H in the picture you showed. Every Pokémon in that game has 3 ability slots. Pokémon that have "one ability" just have the same ability in multiple slots. You can see this easily if you open up a Pokémon like Claydol or Stonjourner in PKHeX and open the drop-down for Ability.
